### PR TITLE
Added customization in RecoveryServices client and updated Microsoft.Rest.ClientRuntime to 2.3.10

### PR DESCRIPTION
**Pull request by sonathan**
**Following items in this PR:**
1. Added customizations in RecoveryServices Client.
2. Updated Microsoft.Rest.ClientRuntime to use 2.3.10

## Description

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md).**
- [x] **The pull request does not introduce [breaking changes](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/Documentation/breaking-changes.md).**

### [General Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#general-guidelines)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#testing-guidelines)
- [ ] Pull request includes test coverage for the included changes.

### [SDK Generation Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#sdk-generation-guidelines)
- [ ] If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.
- [ ] The generate.cmd file for the SDK has been updated with the version of AutoRest, as well as the commitid of your swagger spec or link to the swagger spec, used to generate the code.
- [ ] The `*.csproj` and `AssemblyInfo.cs` files have been updated with the new version of the SDK.

### CODE REVIEW

File: src/SDKs/RecoveryServices/Management.RecoveryServices/Microsoft.Azure.Management.RecoveryServices.csproj [hunk 1]
```diff
@@ -11,6 +11,9 @@
   <PropertyGroup>
     <TargetFrameworks>net452;netstandard1.4</TargetFrameworks>
   </PropertyGroup>
+  <ItemGroup>
+    <PackageReference Update="Microsoft.Rest.ClientRuntime" Version="2.3.10" />
```
Comment (shahabhijeet) [inline src/SDKs/RecoveryServices/Management.RecoveryServices/Microsoft.Azure.Management.RecoveryServices.csproj]: @sonathan this is not needed.
We control the common packages when we have tested and ready to move to a new version of clientruntime throughout the repo.
So when the new version is out and ready to be consumed, this change will be done for the entire repo.
So please revert this change.
If we allow this change, controlling ClientRuntime version across repo will be fragmented and not manageable.

File: src/SDKs/RecoveryServices/Management.RecoveryServices/Customizations/RecoveryServicesClient.cs [hunk 2]
```diff
@@ -39,9 +39,14 @@ public RecoveryServicesClient(System.Uri baseUri, ServiceClientCredentials crede
             DisableDispose = disableDispose;
         }
 
+        public void SetHttpClient(HttpClient client)
```
Comment (shahabhijeet) [inline src/SDKs/RecoveryServices/Management.RecoveryServices/Customizations/RecoveryServicesClient.cs]: FYI
@sonathan the new version 2.3.9 has an overload that will allow you to control disposing of the provided client

**Review by shahabhijeet — CHANGES_REQUESTED** [review]

File: src/SDKs/RecoveryServices/Management.RecoveryServices/Microsoft.Azure.Management.RecoveryServices.csproj (hunk 1 above)
Comment (sonathan) [reply src/SDKs/RecoveryServices/Management.RecoveryServices/Microsoft.Azure.Management.RecoveryServices.csproj]: @Shahabhijeet Thank-you for letting me know of this. I'll update this in next commit.

**Review by shahabhijeet — APPROVED** [review]
